### PR TITLE
Updating tutorials with recent code updates. #189 is part of this work.

### PR DESCRIPTION
**Pull request by DeepBlockDeepak**
### Fixes for: 
* #148 
* #161 
### Remaining/Open TODOS & Issues:
* #165 
* #167

### Fixes for:
* #169
* #170
* #160
### Remaining/Open TODOS & Issues:
* None

### Fixes for:
* None
### Remaining/Open TODOS & Issues:
* None

### Fixes for:
* #170 
* #160 
* #161 
### Remaining/Open TODOS & Issues:
* None

### Fixes for:
* #61 
### Remaining/Open TODOS & Issues:
* #172 

----
:books: Documentation preview :books:: https://pyttb--193.org.readthedocs.build/en/193/

### CODE REVIEW

**Comment by ntjohnson1** [comment]
Thanks for the fast response. Can you cherry-pick or apply the change from here to fix our RTD preview test? https://github.com/sandialabs/pyttb/pull/191 also the cp_apr notebook still uses the old ktensor interface and also needs to be updated.

**Comment by dmdunla** [comment]
Will do …
________________________________
From: Nick ***@***.***>
Sent: Friday, July 7, 2023 2:44:04 PM
To: sandialabs/pyttb ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [EXTERNAL] Re: [sandialabs/pyttb] Updating tutorials with recent code updates. #189 is part of this work. (PR #193)


Thanks for the fast response. Can you cherry-pick or apply the change from here to fix our RTD preview test? #191<https://github.com/sandialabs/pyttb/pull/191> also the cp_apr notebook still uses the old ktensor interface and also needs to be updated.

**Comment by DeepBlockDeepak** [comment]
We *should* be seeing this build error since `cp_apr` is a big computation, correct?:
```
nbclient.exceptions.CellTimeoutError: A cell timed out while it was being executed, after 30 seconds.
The message was: Cell execution timed out.
Here is a preview of the cell contents:
-------------------
# Compute a solution
M = ttb.cp_apr(X, R, printitn = 10);
```

**Comment by dmdunla** [comment]
@DeepBlockDeepak can you change the shape of the tensor to (10, 8, 6). Demonstrating in a smaller problem is fine. If runtime is still too long, we can change params for cp_apr to reduce the runtime.

File: docs/source/tutorial/CP_APR.ipynb [hunk 1]
```diff
@@ -107,7 +100,12 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.8.15"
+   "version": "3.11.2"
+  },
+  "vscode": {
+   "interpreter": {
+    "hash": "b0fa6594d8f4cbf19f97940f81e996739fb7646882a419484c72d19e05852a7e"
+   }
```
Comment (ntjohnson1) [inline docs/source/tutorial/CP_APR.ipynb:108]: @DeepBlockDeepak this vscode meta data appears to break myst_nb. If we look at the runtime from when I first added the preview it takes about 2s. If I remove this vscode snippet locally the build works for me again.

Comment (ntjohnson1) [reply docs/source/tutorial/CP_APR.ipynb:108]: Related to my note earlier about needing to resolve clearing extra information from our notebooks from the repo. If `nbstripout` pulls out the vscode meta data then we might already know the approach. I am kind of surprised this causes things to fall over so hard.

Comment (ntjohnson1) [reply docs/source/tutorial/CP_APR.ipynb]: Hmm! It is actually more interesting. I ran nbstripout and it only changed line ending. I think there is a bad whitespace character that the myst_nb parser is barfing on but regular jupyter figures out.

Comment (ntjohnson1) [reply docs/source/tutorial/CP_APR.ipynb]: I think when I locally removed that snippet when I save it it reflowed the file to linux style LF, where previously it was CRLF

Comment (dmdunla) [reply docs/source/tutorial/CP_APR.ipynb]: @DeepBlockDeepak I think you are developing in Windows, is that correct? Can you check your git line endings and verify that you are not forcing all your commits to use CRLF endings? If this is. It the issue, we should try to determine what is.

Comment (dmdunla) [reply docs/source/tutorial/CP_APR.ipynb]: > Hmm! It is actually more interesting. I ran nbstripout and it only changed line ending. I think there is a bad whitespace character that the myst_nb parser is barfing on but regular jupyter figures out.

nbstripout has option to remove any set of metadata you specify with  —extra-keys parameter. https://github.com/kynan/nbstripout

Comment (DeepBlockDeepak) [reply docs/source/tutorial/CP_APR.ipynb]: I'm developing in MacOS @dmdunla . I did run `git config --get core.autocrlf` and no value was set or returned to me. I'm guessing that's a good thing.

**Comment by dmdunla** [comment]
Maybe VSCode adds those endings?
________________________________
From: DeepBlockDeepak ***@***.***>
Sent: Friday, July 7, 2023 5:09:55 PM
To: sandialabs/pyttb ***@***.***>
Cc: Dunlavy, Daniel M ***@***.***>; Mention ***@***.***>
Subject: [EXTERNAL] Re: [sandialabs/pyttb] Updating tutorials with recent code updates. #189 is part of this work. (PR #193)


@DeepBlockDeepak commented on this pull request.

________________________________

In docs/source/tutorial/CP_APR.ipynb<https://github.com/sandialabs/pyttb/pull/193#discussion_r1256592189>:

> +  "vscode": {
+   "interpreter": {
+    "hash": "b0fa6594d8f4cbf19f97940f81e996739fb7646882a419484c72d19e05852a7e"
+   }


I'm developing in MacOS @dmdunla<https://github.com/dmdunla> . I did run git config --get core.autocrlf and no value was set or returned to me. I'm guessing that's a good thing.

**Pull request merged**